subsys/bluetooth/mesh/prov_device.c

```diff
@@ -280,9 +306,35 @@ static void prov_dh_key_cb(const uint8_t dhkey[32])
 
 	sys_memcpy_swap(bt_mesh_prov_link.dhkey, dhkey, 32);
 
-	BT_DBG("DHkey: %s", bt_hex(bt_mesh_prov_link.dhkey, 32));
+	dh_key_gen_complete();
+}
+
+static void static_dh_key_gen(void)
```

Comment:
Shouldn't it be better to keep all ECC dependent algorithms in crypto.c\.h?

Reply:
It could be moved, but would it make a difference?

Reply:
at the current moment no. However, if someone :) decides to use another security toolbox or to move this code to the crypto memory area it would be done easier.